packages/api/src/platforms/vtex/utils/shouldCreateNewAddress.ts

```diff
@@ -0,0 +1,46 @@
+import { IStoreSession } from '../../../__generated__/schema'
+import { OrderForm } from '../clients/commerce/types/OrderForm'
+
+export const shouldCreateNewAddress = (
+  orderForm: OrderForm,
+  session: IStoreSession
+) => {
+  const availableAddresses = orderForm?.shippingData?.availableAddresses ?? null
```

Comment:
Is available addresses possibly an array? If so, we should check if `availableAdresses.length === 0` on line 9 as well